crates/ruff_python_codegen/src/generator.rs

```diff
@@ -1393,7 +1405,12 @@ impl<'a> Generator<'a> {
 
     fn unparse_string_literal(&mut self, string_literal: &ast::StringLiteral) {
         let ast::StringLiteral { value, flags, .. } = string_literal;
-        self.p_str_repr(value, *flags);
+        let flags = if let Some(quote) = self.quote {
+            flags.with_quote_style(quote)
+        } else {
+            *flags
+        };
+        self.p_str_repr(value, flags);
```

Comment:
I think it's clearer if we move the `self.quote` override to `self.p_str_repr` instead of "patching" the flags here.